### PR TITLE
Add build-image.sh for local build or test image.

### PR DESCRIPTION
**Pull request by xiaojun207**
Add build-image.sh for local build or test image.

### CODE REVIEW

**Comment by hengfeiyang** [comment]
You shouldn't write fix value of params, you can refer: https://github.com/zinclabs/zinc/blob/main/build.sh

**Comment by xiaojun207** [comment]
> 



> You shouldn't write fix value of params, you can refer: https://github.com/zinclabs/zinc/blob/main/build.sh

ok

**Comment by hengfeiyang** [comment]
```
# git describe --tags --always 
v0.2.5-11-g084ebc3
```

Before the version has prefix `v`, So you should remove `v` from `--build-arg VERSION="v${VERSION}"`

And `IMAGE ` has default value `zinclabs/zinc`, i think you want user set it through params instead of change file.
And about `docker push`, default is off, we can provide a param allow user enable it.

File: build-image.sh [hunk 1]
```diff
@@ -1,21 +1,28 @@
...
-COMMIT_HASH=`git rev-parse HEAD`
-IMAGE="zinclabs/zinc"
+VERSION=`git describe --tags --always` # eg.: 0.2.5
+BUILD_DATE=`date +%Y%m%d` # eg.: 20220701
+COMMIT_HASH=`git rev-parse HEAD` #
+IMAGE="$1" #
+
+if [ -z "$1" ]; then
+  IMAGE="zinclabs/zinc"
+fi
```
Comment (hengfeiyang) [inline build-image.sh:13]:
```
IMAGE="zinclabs/zinc"
If [ -n "$1" ]; then
  IMAGE="$1"
fi
```

Do you think this looks better?

**Review by hengfeiyang — APPROVED** [review]

**Pull request merged**